test/sdk_test.py

```diff
@@ -11,56 +11,49 @@
 RED = '\33[31m'
 CLEAR = '\33[0m'
 
-TUNER_LIST = ['TPE', 'Random', 'Anneal', 'Evolution']
+TUNER_LIST = ['BatchTuner', 'TPE', 'Random', 'Anneal', 'Evolution']
+ASSESSOR_LIST = ['Medianstop']
 EXPERIMENT_URL = 'http://localhost:8080/api/v1/nni/experiment'
 
 
-def switch_tuner(tuner_name):
-    '''Change tuner in config.yml'''
-    config_path = 'sdk_tuner_test/local.yml'
+def switch(dispatch_type, dispatch_name):
+    '''Change dispatch in config.yml'''
+    config_path = 'sdk_test/local.yml'
     experiment_config = get_yml_content(config_path)
-    experiment_config['tuner'] = {
-        'builtinTunerName': tuner_name,
+    experiment_config[dispatch_type.lower()] = {
+        'builtin' + dispatch_type + 'Name': dispatch_name,
         'classArgs': {
             'optimize_mode': 'maximize'
         }
     }
     dump_yml_content(config_path, experiment_config)
 
-def test_builtin_tuner(tuner_name):
-    remove_files(['sdk_tuner_test/nni_tuner_result.txt'])
-    switch_tuner(tuner_name)
+def test_builtin_dispatcher(dispatch_type, dispatch_name):
+    switch(dispatch_type, dispatch_name)
 
-    print('Testing %s...'%tuner_name)
-    proc = subprocess.run(['nnictl', 'create', '--config', 'sdk_tuner_test/local.yml'])
+    print('Testing %s...' % dispatch_name)
+    proc = subprocess.run(['nnictl', 'create', '--config', 'sdk_test/local.yml'])
     assert proc.returncode == 0, '`nnictl create` failed with code %d' % proc.returncode
 
     nnimanager_log_path = fetch_experiment_config(EXPERIMENT_URL)
     
     for _ in range(10):
         time.sleep(3)
-
-        # check if tuner exists with error
-        tuner_status = read_last_line('tuner_result.txt')
-        assert tuner_status != 'ERROR', 'Tuner exited with error'
-
         # check if experiment is done
         experiment_status = check_experiment_status(nnimanager_log_path)
         if experiment_status:
             break
     
     assert experiment_status, 'Failed to finish in 30 sec'
 
-def run():
-    to_remove = ['tuner_search_space.json', 'tuner_result.txt', 'assessor_result.txt']
-    remove_files(to_remove)
-
-    for tuner_name in TUNER_LIST:
+def run(dispatch_type):
+    LIST = TUNER_LIST if dispatch_type == 'Tuner' else ASSESSOR_LIST
```

Comment:
You assume here dispatch_type SHOULD be 'Assessor' if it's not 'Tuner', but I suggest we should also check if dispatch_type is 'Assessor' if it's not 'Tuner', and throw error if the validation failed.